src/patchelf.cc

```diff
@@ -1885,9 +1906,186 @@ void ElfFile<ElfFileParamNames>::clearSymbolVersions(const std::set<std::string>
         }
     }
     changed = true;
-    this->rewriteSections();
+    rewriteSections();
 }
 
+struct symstr {
+    const char *str;
+    size_t len;
+};
+
+// Elf32_Off, Elf32_Word, Elf54_Word = unsigned int
+// Elf64_Off, Elf64_Xword = unsigned long (or long long on 32 bit)
+//typedef variant<Elf32_Off*,Elf32_Word*,Elf64_Off*,Elf64_Word*,Elf64_Xword*> symref;
+//typedef variant<unsigned int*, unsigned long*> symref;
+typedef variant<uint16_t*, uint32_t*, uint64_t*> symref;
+
+struct Comparator {
+    using is_transparent = void;
+    bool operator()(const symstr& a, const symstr& b) const
+    {
+        return (a.str+a.len < b.str+b.len);
+    }
+
+    bool operator()(const char* a, const symstr& b) const
+    {
+        return (a < b.str);
+    }
+
+    bool operator()(const symstr& a, const char* b) const
+    {
+        return (a.str+a.len < b);
+    }
+};
+
+#include <argz.h>
+
+class mymap : public map<symstr, forward_list<symref>, Comparator> {
+
+    public:
+	size_t totalsize;
+    mymap(const char *s, size_t s_length) : totalsize(s_length)
+    {
+        const char *c = NULL;
+        while ((c=argz_next(s, s_length, c))) {
+            insert({(const struct symstr){c, strlen(c)},forward_list<symref>()});
+        }
+    }
+
+    forward_list<symref>& operator[](const char *i)
+    {
+        return lower_bound(i)->second;
+    }
+
+    template <class T>
+    void incref(const char *idx, T *sym)
+    {
+    	(*this)[idx].push_front(sym);
+    }
+
+};
+
+template<ElfFileParams>
+void ElfFile<ElfFileParamNames()>::cleanstrtab()
+{
+    auto shdrDynamic = findSectionHeader(".dynamic");
+    auto shdrDynStr = findSectionHeader(".dynstr");
+    auto shdrDynSym = findSectionHeader(".dynsym");
+
+    auto strTab = (char *)fileContents->data() + rdi(shdrDynStr.sh_offset);
+    int verneednum = 0;
+    int verdefnum = 0;
+    //mkref(strTab, rdi(shdrDynStr.sh_size)); // new strInfo()
```

Comment:
cleanup old code

Reply:
weird, this line is gone already here.... commit "remove some leftover comment-out lines " should have removed it.